### ISSUE TITLE
more realistic guns charging/discharging

### ISSUE THREAD

**Issue by callmepartario** [heclak/community-a4e-c#503]
from user MK-12-20MM#4382 on discord:

`Nice touch on the slat sounds.  This reminds me that there should be a sound for charging the MK-12 IMHO. Background is that we would charge the nitrogen bottle to MAX which would be 3000 PSI for gun sorties.  Testing the MK-12 on the ground and charging the gun was rather a large twang. Clearing (SAFE) the bolt would bang against the buffer. (just a note: we had test cartridges primer (electrical) only-to ensure firing pin contact).  With a generator running you could hear (and feel) the charging valve cycle -> and the bolt slamming forward to battery even in the cockpit. Now I don't know from a in-flight perspective, but since there is sound for the slats and gun charging is a violent affair.. Scary as hell if your are next to the next the thing.  Not a big deal but it might be a nice touch in the future. Unless someone could in-flight (A SME) confirm there was no sound on Gun Charging. I never asked a pilot when I worked on this bird a billion years ago..   Thanks...`

**Comment by MK12-20MM** [comment]
I was going to raise this issue/enhancement but waiting on confirmation from my own query to a Scooter driver.  Thanks regardless.

**Comment by callmepartario** [comment]
SME consult dispatch:
- guns are ejected by nitrogen pressure. the sound is a quick, low, thud, which is more felt than heard.
- the sound is only heard once per flight - once the guns fall into position, they remain there.
- setting the guns to SAFE has a repeatable sound of ejecting the last round or two - the sound is similar, but should happen ~0.5m behind. both sounds are fundamentally the breech-mechanism of the gun.

**Comment by callmepartario** [comment]
- Sound programming added in https://github.com/heclak/community-a4e-c/commit/d64ceb4919ead84096b02af0c5f040af2a402517
- Binaries are updated with new two assets
- SMEs confirm that the guns really should only arm and safe ONCE per mission. I've not updated the code for this behavior yet,  but will before this issue is closed.

**Comment by callmepartario** [comment]
I've added some implementation of both the sound and a single pair of ARM/SAFE procedures in https://github.com/heclak/community-a4e-c/commit/a91b8bf4c90da97dd1f1ac02a317eca140c3ea9c

Here is a quick rundown of the new behaviors for documentation in the kneeboard:
- There is enough nitrogen pressure to charge and safe the guns ONCE per sortie.
- When the gear is up and power is available, the Guns switch can arm the guns (you will hear the breech slide into place).
- In order to fire the guns using the Gun-Rocket Trigger, the guns much be charged, and the Master Armament Switch must be in the ON position.
- While charged, returning the guns switch to SAFE, putting the gear down, or losing AC power, will interrupt the gun charge and safe the guns (you will hear the breech slide back out of place).
- When the ground crew re-arms the aircraft, they will reset any unsafe guns and restore any nitrogen charge needed to arm them back to the maximum.

**Comment by callmepartario** [comment]
audio notes: 
- best to set these sounds to separate L/R placements to accommodate head turning.
- an array of 2 or 4 sounds should suffice, each potential pairing should be selected to accommodate the maximum possible phase cancellation.

**Comment by callmepartario** [comment]
assets are hooked and finalized, closing this issue.

**Comment by MK12-20MM** [comment]
There is no guaranteed that the nitrogen system has not depleted. Depending on the A/C there might be slow leaks. In any case no attempt should be made after clearing the gun to recharge the gun as there might be a round stuck in the chamber. If HEI/API rounds are used then attempting the to re-charge the gun could cause an even bigger problem. Typically if this occurred (stuck round) we would remove the gun at the EOR (Arm/DeArm) and take it to EOD to verify if the round had fired or removing the barrel if the round had not been fired for appropriate disposal.

I have another comment about statements made: 

"While charged, returning the guns switch to SAFE, putting the gear down, or losing AC power, will interrupt the gun charge and safe the guns (you will hear the breech slide back out of place) "

I disagree with some of it. Putting the gear down or weight on wheels, loss of A/C will inhibit the guns from firing but will not clear the gun as in sending the breechblock back.   For example, if there was a round in the chamber and as stated in the comment that putting the gear down will clear (breechblock back) such action would extract the round (if the round is not stuck) and drop it from the Aircraft. Potential for Dropping live rounds over populated areas is a safety issue. 

Loss of AC power will prevent the gun from firing, but this action is un-commanded and will not send the breechblock back as again, if there was a round in the chamber sending breechblock back will extract the round (if not stuck) and drop it from the aircraft. Potential for Dropping live rounds over populated areas is a safety issue. In fact dropping a fired case would be an issue as well.

I think there is some confusion with some of the information given.

**Comment by callmepartario** [comment]
> 

you sound super-knowledgeable! are you on our discord? i was a little brief in these notes vs. exact simulated dependencies, but i would love to talk with you further on the subject.

**Comment by callmepartario** [comment]
- when the aircraft spawns or is rearmed after using the guns, a gun reliablity rating is represented by a random even number from 2 to 10.
- arming or safing the guns reduces this reliability rating by 1.
- if the reliability rating of the gun is less than 2, the guns cannot be armed (so that the final charge is always expended by safing the gun.

additionally the following behaviors have been revised:
- the gear being down now has no bearing on gun arming or safing. you can now take off and land with armed guns, however, they cannot be fired due to the master armament being disabled in this state.
- a loss of AC power no longer safes the guns. If power loss occurs while the guns are armed, they can still be safed by moving the switch.

**Comment by MK12-20MM** [comment]
I dropped off the discord.. I can jump back on if you want.  Most of the time  partial firing sequence was due to the ammo belt binding in the can or link chute which could be contributed or suspected to ACM.  Based on my experience working on the MK-12 system would probably 80-90 percent success rate of firing a whole belt for each gun. An actual round jam, where the round was stuck in the chamber and did not extract was rare but did happen on occasion.
